### PR TITLE
chore: Upgrade okta-auth-js in okta-signin-widget 3.0

### PR DESCRIPTION
**Pull request by manueltanzi-okta**
### Description

- Upgrade okta-auth-js in okta-signin-widget 3.0
- Use cross-fetch auth-js version instead of jquery one

**Note:** Will go in okta-signin-widget v3.0.0.

### Reviewers

@gowthamidommety-okta @robertdamphousse-okta

### CODE REVIEW

File: src/util/Util.js [hunk 1]
```diff
@@ -61,12 +61,11 @@ define(['okta', './Logger', './Enums'], function (Okta, Logger, Enums) {
...
     }
     if (!xhr.responseJSON) {
-      try {
-        xhr.responseJSON = JSON.parse(xhr.responseText);
-      } catch (parseException) {
+      if (!xhr.responseText) {
         xhr.responseJSON = { errorSummary: Okta.loc('oform.error.unexpected', 'login') };
         return xhr;
       }
+      xhr.responseJSON = xhr.responseText;
```
Comment (manueltanzi-okta) [inline src/util/Util.js]: **Note:** The logic of how okta-auth-js returns errors changed a little bit in the new version, this is why this change. The new version of okta-auth-js return the responseText already parsed from JSON, so we don’t need to parse it again here (which would fail, since they are not anymore in JSON format). We need instead to check if that attribute is available.

File: src/widget/OktaSignIn.js [hunk 2]
```diff
@@ -95,7 +95,7 @@ var OktaSignIn = (function () {
   function OktaSignIn (options) {
     require('okta');
 
-    var OktaAuth = require('@okta/okta-auth-js/jquery');
+    var OktaAuth = require('@okta/okta-auth-js');
```
Comment (manueltanzi-okta) [inline src/widget/OktaSignIn.js]: **Note:** New version of okta-auth-js uses `cross-fetch` by default! (https://github.com/okta/okta-auth-js/pull/199), which is we want to use cross-fetch in version 3.0. So we don't need to specify `/jquery` version anymore.

(This has been already cross-browser + embedded browser tested for the widget when we made the change in okta-auth-js).

File: test/unit/spec/PrimaryAuth_spec.js [hunk 3]
```diff
@@ -1920,36 +1920,6 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthForm, Be
             expectErrorEvent(test, 429, 'API call exceeded rate limit due to too many requests.');
           });
       });
-      itp('shows an error if authClient returns with an error that is plain text', function () {
```
Comment (gowthamidommety-okta) [inline test/unit/spec/PrimaryAuth_spec.js]: Why are we removing this test? authClient could still return an error that is plain text right?

Comment (manueltanzi-okta) [reply test/unit/spec/PrimaryAuth_spec.js]: They authClient should always return the parsed object, while in this tests we were mocking that returns a string, which we don't parse anymore in the new way and which should throw this error: https://github.com/okta/okta-signin-widget/pull/692/files#diff-a2204dc4da481825a3d452cdd6c378ccR65.

But good point, I should add test for that 👍
